### DISCUSSION TITLE
测试特殊字符无法导入

### DISCUSSION THREAD

**Discussion by lm520hy** [apache/doris#23449] [RFC]
语法：cat customer.txt | curl --location-trusted -u root:  -H "timeout:1200" -H "column_separator:|" -T -  http://192.168.194.138:8030/api/tpcds/customer/_stream_load
无法导入的文本格式
322|AAAAAAAACEBAAAAA|367590|256|93369|2449888|2449858|Miss|Sunshine|Solomon|Y|15|2|1928|RÉUNION||Sunshine.Solomon@KqIOuo0jiM9KN.com|2452392|
里面的特殊字符是É
报错
Reason: Unable to display. src line [Unable to display]; 
![Q}CKU%_9$6E DYOKY 1Z(G3](https://github.com/apache/doris/assets/29275555/8243169c-21e1-48dd-a74b-56728c55d25d)
![EM61}~DXDHEA_FJH4EOQ6RY](https://github.com/apache/doris/assets/29275555/9c83ac0b-78e8-4591-9454-2dac62052595)

**Comment by ixzc** [comment]
Hello, what is your version number, please? I tested it on 2.0.0 release and 1.2.6 release, but I didn't find the problem you mentioned.、
<img width="749" alt="image" src="https://github.com/apache/doris/assets/49646212/e80036be-1dbc-4e86-8760-d3dcee7a47b4">
<img width="191" alt="image" src="https://github.com/apache/doris/assets/49646212/bcfa1f8e-696f-4f8f-bcb4-20a39378b5b5">

**Reply by lm520hy** [reply]
你好
我的版本是：版本是apache-doris-2.0.0-bin-x64.tar.gz
数据来源是：TPC-DS自动生成的（处理后去掉最后"|"）
测试中发现直接复制出来，使用echo 可以导入数据，但是文本不可以导入
我的建表语句是
create table customer
(
    `c_customer_sk`             INT               not null,
    `c_customer_id`             char(16)              not null,
    `c_current_cdemo_sk`        INT                       ,
    `c_current_hdemo_sk`        INT                       ,
    `c_current_addr_sk`         INT                       ,
    `c_first_shipto_date_sk`    INT                       ,
    `c_first_sales_date_sk`     INT                       ,
    `c_salutation`              char(10)                      ,
    `c_first_name`              char(20)                      ,
    `c_last_name`               char(30)                      ,
    `c_preferred_cust_flag`     char(1)                       ,
    `c_birth_day`               INT                       ,
    `c_birth_month`             INT                       ,
    `c_birth_year`              INT                       ,
    `c_birth_country`           varchar(20)                   ,
    `c_login`                   char(13)                      ,
    `c_email_address`           char(50)                      ,
    `c_last_review_date`        char(10)                      
)
DUPLICATE KEY(`c_customer_sk`)
DISTRIBUTED BY HASH(`c_customer_sk`) BUCKETS 1
PROPERTIES (
    "replication_allocation" = "tag.location.default: 1"
);


![1](https://github.com/apache/doris/assets/29275555/46542371-0677-4409-8887-5ddebbd596a8)
![2](https://github.com/apache/doris/assets/29275555/1ca89351-17a3-4994-a05e-beaf08f9ea3c)
![3](https://github.com/apache/doris/assets/29275555/3e0b1580-b1f5-40da-a491-c7763c6e05b0)
[customer01.txt](https://github.com/apache/doris/files/12436322/customer01.txt)

**Comment by ixzc** [comment]
Hello, for this question. First, you'd better take a look at the error link, which will give you the details. Based on the syntax you provided, the test was repeated. The following is the error link:
<img width="1380" alt="image" src="https://github.com/apache/doris/assets/49646212/08037f30-3227-4670-a92a-67f8ba501e98">
<img width="1555" alt="image" src="https://github.com/apache/doris/assets/49646212/49997a70-925e-4c1a-83ff-65604091f0ec">

In addition, I found new problems: column_name[c_login], the length of input is too long than schema. You should modify the corresponding fields.
<img width="1544" alt="image" src="https://github.com/apache/doris/assets/49646212/28f47320-234c-4b18-881d-2cc48e7929c0">

**Reply by lm520hy** [reply]
你好
我最后一张发出来的图片中有报错信息如下：
![1692970167202](https://github.com/apache/doris/assets/29275555/a1bf00b4-4859-4471-91d0-f0cbc2a7b833)
是使用的curl打印出来的，并且我的错误和你并不一致
关于你说的字段[c_login]的位置不对我使用EXCLE对应起来如下：
![001109166ab952a8cfea9d57aeff01a](https://github.com/apache/doris/assets/29275555/c806dd22-71a1-4512-9f6c-186a82e72371)
并且我试过使用varchar替代char来作为类型进行导入依然报错

**Reply by ixzc** [reply]
can you add my wechat ? (17625672480) So I can better respond to your question.

**Comment by zhbdesign** [comment]
用json类型应该可以，我们是遇到特殊符号的就转为json，但是默认的csv性能是最好的

**Reply by lm520hy** [reply]
周一去试一下 电脑没带回来  主要是报错直接退出了 使用json中间多了一些过程势必会影响性能
另外测试一下其他库这样数据使用seaturn导入会报错吗

**Reply by zhbdesign** [reply]
如果不是并发大量数据，json的性能也可以接受吧
